org.eclipse.m2e.jdt.tests/src/org/eclipse/m2e/jdt/tests/JpmsConfigurationTest.java

```diff
@@ -0,0 +1,163 @@
+    assertTrue(!jreAttributes.containsKey(PATCH_MODULE_ATTR));
+    
+    assertTrue(!m2eAttributes.containsKey(ADD_EXPORTS_ATTR));
+    assertTrue(!m2eAttributes.containsKey(ADD_OPENS_ATTR));
+    assertTrue(!m2eAttributes.containsKey(ADD_READS_ATTR));
+    assertTrue(!m2eAttributes.containsKey(PATCH_MODULE_ATTR));
+    
+    
+    
+    String contents = read(project, pomFileFS);
+    String argsSet1 = read(project, argsSet1FS);
+    String argsSet2 = read(project, argsSet2FS);
+    
+    IFile pomFileWS = project.getFile(pomFileFS.getName());
+    
+    String argsSet1Content = contents.replace(REPLACED_POM_STRING, argsSet1);
+    pomFileWS.setContents(new ByteArrayInputStream(argsSet1Content.getBytes()), true, false, null);
+    waitForJobsToComplete();
+    
+    jreAttributes = getJreContainerAttributes(javaProject);
+    m2eAttributes = getM2eContainerAttributes(javaProject);
+    
+    assertTrue(jreAttributes.get(ADD_EXPORTS_ATTR).equals(JRE_ADD_EXPORTS_VALUE1));
+    assertTrue(jreAttributes.get(ADD_OPENS_ATTR).equals(JRE_ADD_OPENS_VALUE1));
+    assertTrue(jreAttributes.get(ADD_READS_ATTR).equals(JRE_ADD_READS_VALUE1));
+    assertTrue(jreAttributes.get(PATCH_MODULE_ATTR).equals(JRE_PATCH_MODULE_VALUE1));
+    
+    assertTrue(m2eAttributes.get(ADD_EXPORTS_ATTR).equals(M2E_ADD_EXPORTS_VALUE1));
+    assertTrue(m2eAttributes.get(ADD_OPENS_ATTR).equals(M2E_ADD_OPENS_VALUE1));
+    assertTrue(m2eAttributes.get(ADD_READS_ATTR).equals(M2E_ADD_READS_VALUE1));
+    assertTrue(m2eAttributes.get(PATCH_MODULE_ATTR).equals(M2E_PATCH_MODULE_VALUE1));
+    
+    String argsSet2Content = contents.replace(REPLACED_POM_STRING, argsSet2);
+    pomFileWS.setContents(new ByteArrayInputStream(argsSet2Content.getBytes()), true, false, null);
+    waitForJobsToComplete();
+    
+    jreAttributes = getJreContainerAttributes(javaProject);
+    m2eAttributes = getM2eContainerAttributes(javaProject);
+    
+    assertTrue(jreAttributes.get(ADD_EXPORTS_ATTR).equals(JRE_ADD_EXPORTS_VALUE1 + ATTR_VALUE_SEPARATOR + JRE_ADD_EXPORTS_VALUE2));
+    assertTrue(jreAttributes.get(ADD_OPENS_ATTR).equals(JRE_ADD_OPENS_VALUE1 + ATTR_VALUE_SEPARATOR + JRE_ADD_OPENS_VALUE2));
+    assertTrue(jreAttributes.get(ADD_READS_ATTR).equals(JRE_ADD_READS_VALUE1 + ATTR_VALUE_SEPARATOR + JRE_ADD_READS_VALUE2));
+    assertTrue(jreAttributes.get(PATCH_MODULE_ATTR).equals(JRE_PATCH_MODULE_VALUE1 + ATTR_VALUE_SEPARATOR + JRE_PATCH_MODULE_VALUE2));
+    
+    assertTrue(m2eAttributes.get(ADD_EXPORTS_ATTR).equals(M2E_ADD_EXPORTS_VALUE1 + ATTR_VALUE_SEPARATOR + M2E_ADD_EXPORTS_VALUE2));
+    assertTrue(m2eAttributes.get(ADD_OPENS_ATTR).equals(M2E_ADD_OPENS_VALUE1 + ATTR_VALUE_SEPARATOR + M2E_ADD_OPENS_VALUE2));
+    assertTrue(m2eAttributes.get(ADD_READS_ATTR).equals(M2E_ADD_READS_VALUE1 + ATTR_VALUE_SEPARATOR + M2E_ADD_READS_VALUE2));
+    assertTrue(m2eAttributes.get(PATCH_MODULE_ATTR).equals(M2E_PATCH_MODULE_VALUE1 + ATTR_VALUE_SEPARATOR + M2E_PATCH_MODULE_VALUE2));
+  }
+  
```

Comment:
would be nice to cover testCompilerArguments as well

Reply:
Test now cover testCompilerArguments.

Reply:
Removed testCompilerArguments half backed support. I did try to handle the case but all in all it is not possible.
testCompilerArguments does not handle double dashed arguments so do i.